inputremapper/configs/mapping.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# input-remapper - GUI for device specific keyboard mappings
+# Copyright (C) 2022 sezanzeb <proxima@sezanzeb.de>
+#
+# This file is part of input-remapper.
+#
+# input-remapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# input-remapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with input-remapper.  If not, see <https://www.gnu.org/licenses/>.
+
+import enum
+from evdev.ecodes import EV_KEY
+from pydantic import BaseModel, PositiveInt, confloat, root_validator, validator
+from typing import Optional
+
+from inputremapper.key import Key
+from inputremapper.configs.system_mapping import system_mapping
+from inputremapper.injection.macros.parse import is_this_a_macro, parse
+
+
+# TODO: in python 3.11 use enum.StrEnum
+class KnownUinput(str, enum.Enum):
+    keyboard = "keyboard"
+    mouse = "mouse"
+    gamepad = "gamepad"
+
+
+class Mapping(BaseModel):
```

Comment:
How do you feel about an added dependency: https://pydantic-docs.helpmanual.io/
I think it is a perfect fit for managing all configs which a user might change.

It provides automatic type casting and validation for variable assignment, and enforces the variable types defined in class definition.

I just implemented the Mapping class to see how it would work. I think the benefits are huge, and I could remove all the type and value validation inside the `MappingHandler`
```python
from pydantic import BaseModel, confloat
class Mapping(BaseModel):
    deadzone: confloat(ge=0, le=1) = 0.1

    class Config:
        validate_assignment = True

mapping = Mapping(deadzone="0.5")  # will automatically cast to float
mapping.deadzone = -1  # will raise an ValidationError: deadzone must be greater or equal to 0
```

Reply:
it seems to be available in both ubuntu and arch repos, sounds good to me